### ISSUE TITLE
Issue with collections dependency while running GitHub Actions

### ISSUE THREAD

**Issue by narko** [jarekb123/dart_dataclass#8]
Because every version of flutter_test from sdk depends on collection 1.15.0-nullsafety and dataclass 0.3.0+1 depends on collection >=1.14.0 <1.15.0, flutter_test from sdk is incompatible with dataclass 0.3.0+1.

And because no versions of dataclass match >0.3.0+1 <0.4.0, flutter_test from sdk is incompatible with dataclass ^0.3.0+1.

**Comment by jarekb123** [comment]
fixed with version 0.5.0

**Comment by erickentry** [comment]
Still happens in `0.5.0`:
```
Because every version of flutter_test from sdk depends on collection 1.15.0-nullsafety.3 and dataclass >=0.3.0 depends on collection >=1.14.0 <1.15.0, flutter_test from sdk is incompatible with dataclass >=0.3.0.
So, because ******** depends on both dataclass ^0.5.0 and flutter_test any from sdk, version solving failed.
```
